Dockerfile Update: Add conditional db seeding to `init.dev.sh`

neenjaw opened the pull request:
Close #180

neenjaw committed:
Add conditional db seeding to `init.dev.sh`

iHiD commented on docker/dev/init.dev.sh:
```diff
@@ -12,5 +12,11 @@ echo "Create and migrate DBs"
 mysql -u root -ppassword -h "$DB_HOST" < docker/dev/init.sql
 bundle exec bin/rails db:migrate
 
+num_users = $(rails r "p User.count")
```
```suggestion
num_users = $(bundle exec rails r "p User.count")
```

neenjaw committed:
Update docker/dev/init.dev.sh

Co-authored-by: Jeremy Walker <jez.walker@gmail.com>

neenjaw commented on docker/dev/init.dev.sh:
```diff
@@ -12,5 +12,11 @@ echo "Create and migrate DBs"
 mysql -u root -ppassword -h "$DB_HOST" < docker/dev/init.sql
 bundle exec bin/rails db:migrate
 
+num_users = $(bundle exec rails r "p User.count")
```
```suggestion
num_users=$(bundle exec rails r "p User.count")
```

neenjaw committed:
Update docker/dev/init.dev.sh

neenjaw committed:
Updated

neenjaw committed:
shell script changed for bourne shell
Script was previously written for bourne again shell (bash)
where the docker image calls `init.dev.sh` using `sh` rather than `bash`
So it was failing at the `if` comparison since `[[ ... ]]` only exists
in `bash`.
